### DISCUSSION TITLE
Interface network_name usage and intent

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#381] [Help & Questions]
According to the docs network_name creates separate logical networks on the same physical hardware. 

My assumption is that this is for interfaces on the same node. Example: if you had five interfaces, two named NetworkA and two named NetworkB, the named networks would be linked (assuming transport is enabled) across both interfaces, even if the destination's interface was names differently or not at all. 

Will any named interface not bridge to any unnamed interface?

It makes sense to operate this way, but I think building a test network would be far more work than just asking, so I figured I'd ask. :D 

I mean, I assume if both NetworkA and NetworkB linked to a common node they'd end up linked that way, but that's just how networks . . . work.

**Comment by markqvist** [comment]
Yes, that's correct. You can think of the network name parameter a bit like how VLANs work for Ethernet. The implementation is very different (since it uses the IFAC system, instead of a simple numerical tag in the frame header), but the result is functionally the same in most regards: You can run multiple separate virtual networks over the same physical medium.

Now, you can of course link these networks back together, if you have a transport node providing connectivity between the different segments. So it's very much a tool, without any *one* specific intended purposes, but intended to allow users more flexibility in how they configure and structure their networks.
